### PR TITLE
ENH multi-level design support

### PR DESCRIPTION
**Pull request by BorisMuzellec**
#### Reference Issue or PRs

Related to #100 

#### What does your PR implement? Be specific.

This PR extends multi-factor analysis from bi-level to n-level designs. Continuous variables are left to future work.

### CODE REVIEW

File: tests/test_pydeseq2.py [hunk 1]
```diff
@@ -255,7 +255,7 @@ def test_multifactor_lfc_shrinkage(tol=0.02):
     res = DeseqStats(dds)
     res.summary()
     res.SE = r_res.lfcSE * np.log(2)
-    res.lfc_shrink()
```
Comment (maikia) [inline tests/test_pydeseq2.py:258]: can you also test that if the wrong coeff is being passed there is an error raised

**Review by maikia — APPROVED** [review]
LGTM, thanks @BorisMuzellec

File: pydeseq2/ds.py [hunk 2]
```diff
@@ -544,3 +512,76 @@ def objective(a: float) -> float:
             return min_var
         else:
             return root_scalar(objective, bracket=[min_var, max_var]).root
+
+    def _build_contrast(self, contrast: Optional[List[str]] = None) -> None:
+        """
+        Check the validity of the contrast (if provided).
```
[the remark this answers is not in the document]
Comment (BorisMuzellec) [reply pydeseq2/ds.py]: Done in [d2a2fa6](https://github.com/owkin/PyDESeq2/pull/101/commits/d2a2fa6507e05e8174977c918f02c85422835afb). Also added tests

File: tests/test_pydeseq2.py (hunk 1 above)
Comment (BorisMuzellec) [reply tests/test_pydeseq2.py:258]: Added in [9b73f0c](https://github.com/owkin/PyDESeq2/pull/101/commits/9b73f0cc1326ff3155eb20035a1574a0125a2b2d)

**Pull request merged**